packages/component-library/src/components/plmg-separator/plmg-separator.tsx

```diff
@@ -0,0 +1,95 @@
+import { Component, h, Prop, Watch } from '@stencil/core';
+import { plmgColorBorderNeutral } from '@ducky/plumage-tokens';
+
+import {
+  isPlmgSeparatorDirection,
+  isPlmgSeparatorThickness,
+} from './plmg-separator.types';
+
+@Component({
+  tag: 'plmg-separator',
+  styleUrl: 'plmg-separator.scss',
+  shadow: true,
+})
+export class Separator {
+  /**
+   * Define icon's color.
+   *
+   * Can be any valid CSS color value.
+   *
+   * Default is plmgColorBorderNeutral
+   */
+
+  @Prop() color: string = plmgColorBorderNeutral;
+  @Watch('color')
+  validateColor(newValue: string) {
+    if (typeof newValue !== 'string' || newValue === '')
+      throw new Error('color: must be a valid value');
+  }
+  /**
+   * Define separator's direction.
+   *
+   * Allowed values:
+   *   - vertical
+   *   - horizontal
+   *
+   * Default: horizontal
+   */
+
+  @Prop() direction: string = 'horizontal';
+  @Watch('direction')
+  validateDirection(newValue: string) {
+    if (typeof newValue !== 'string' || newValue === '')
+      throw new Error('direction: required');
+    if (!isPlmgSeparatorDirection(newValue))
+      throw new Error('direction: must be a valid value');
+  }
+  /**
+   * Define separator's thickness.
+   *
+   * Allowed values:
+   *   - thin
+   *   - thick
+   *
+   * Default: thin
+   */
+  @Prop() thickness: string = 'thin';
+  @Watch('thickness')
+  validateThickness(newValue: string) {
+    if (typeof newValue !== 'string' || newValue === '')
+      throw new Error('thickness: required');
+    if (!isPlmgSeparatorThickness(newValue))
+      throw new Error('thickness: must be a valid value');
+  }
+
+  render() {
+    const verticalClasses = {
+      'plmg-separator-container-vertical': true,
+      [this.thickness]: true,
+      [this.direction]: true,
+    };
+
+    const seperatorClasses = {
+      'plmg-separator': true,
+      [this.thickness]: true,
+      [this.direction]: true,
+    };
+
+    const backgroundColorStyle = {
+      backgroundColor: this.color ?? plmgColorBorderNeutral,
+    };
+
+    return (
+      <div
+        class={
+          this.direction === 'vertical'
+            ? verticalClasses
+            : 'plmg-separator-container-horizontal'
+        }
+        style={this.direction === 'vertical' && backgroundColorStyle}
+      >
+        <hr class={seperatorClasses} style={backgroundColorStyle} />
+      </div>
```

Comment:
I'm finding your logic here confusing. 
You are applying the same type of styles in two different ways depending on the `direction` prop. For example, applying background colour on the div and hr does not seem necessary. Only the hr should need the colour.

Reply:
```suggestion
<div class={`plmg-separator-container-${this.direction}`}>
  <hr class={seperatorClasses} style={backgroundColorStyle} />
</div>
```
Upon some playing about in storybook this is a much more consistent approach to applying the styles. A stylesheet change is required see next comment.

Reply:
I managed to convince myself that the div required a background color on vertical direction. I thought this was the solution to the problem of the component failing to work within another component using flexbox. Thanks for the simplification.